todo_app/data/trello_items.py

```diff
@@ -0,0 +1,95 @@
+import requests,os, json
+from dotenv import load_dotenv, find_dotenv
+
+load_dotenv(find_dotenv())
+
+key = os.getenv('TRELLO_API_KEY') 
+token = os.getenv('TRELLO_API_TOKEN')
+
+
+
+class Item:
+    def __init__(self, id, name, status, desc):
+        self.id = id
+        self.name = name
+        self.status = status
+        self.desc = desc 
+
+    @classmethod
+    def from_trello_card(cls, card, lists):
+        list_name = next((list['name'] for list in lists if list['id'] == card['idList']), None)
+        return cls(card['id'], card['name'], list_name, card.get('desc', ''))
+
+def get_board_id():
+    fields = 'fields=name'
+    url = 'https://api.trello.com/1/members/me/boards?{}&key={}&token={}'.format(fields,key,token)
+    reponse = requests.get(url)
+    boards_json = reponse.json()
+
+    for board in boards_json:
+        if board.get('name') =='DevOps Engineering':
+            board_id = board.get('id')
+    return board_id
+
+def get_lists():
+    fields = 'fields=name'
+    url = 'https://api.trello.com/1/boards/{}/lists?{}&key={}&token={}'.format(get_board_id(),fields,key,token)
+    reponse = requests.get(url)
+    lists_json = reponse.json()
+    return lists_json
+
+
+def get_items():
+    items = []
+    fields = 'fields=name,idList,desc,labels'
+    url = 'https://api.trello.com/1/boards/{}/cards?{}&key={}&token={}'.format(get_board_id(),fields,key,token)
+    reponse = requests.get(url)
+    cards_json = reponse.json()
+
+    for card in cards_json:
+        for list in get_lists():
```

Comment:
Your app is a little slow and I suspect this is why!
Every single time you look at a card, you make a fresh request to Trello, so if you have 5 cards in your board, you'll get the same info from Trello at least 5 times - but that gives us a nice easy fix at least - it seems considerably faster for me if I switch this for:
```python
    lists = get_lists()

    for card in cards_json:
        for list in lists:
```

Reply:
I have made this change & the improvement on performance is significant :)